### PR TITLE
Update cache guide to including the caching for traced models

### PR DESCRIPTION
**Pull request by JingyaHuang**
# What does this PR do?

Add doc

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: docs/source/guides/cache_system.mdx [hunk 1]
```diff
@@ -12,24 +12,26 @@ specific language governing permissions and limitations under the License.
...
 The Neuron Model Cache is hosted on the [Hugging Face Hub](https://huggingface.co/aws-neuron/optimum-neuron-cache) and includes compiled files for all popular and supported `optimum-neuron` pre-trained models.
 
-Before training a Transformers or Diffusion model or loading a NeuronModelForCausalLM on Neuron platforms, it needs to be exported to neuron format
+Before training a Transformers or Diffusion model or loading a NeuronModelForXXX on Neuron platforms, it needs to be exported to neuron format
 with [`torch-neuronx`](https://github.com/aws-neuron/aws-neuron-samples/tree/master/torch-neuronx).
 
-When exporting a model, [`torch-neuronx`](https://github.com/aws-neuron/aws-neuron-samples/tree/master/torch-neuronx) will:
+When exporting a model, [`torch-neuronx`](https://github.com/aws-neuron/aws-neuron-samples/tree/master/torch-neuronx) will either:
+
+- 1st approach: convert it to a set of [XLA](https://github.com/pytorch/xla/) subgraphs, and compile each subgraph with the neuronx compiler into a Neuron Executable File Format (NEFF) binary file(eg. training and inference of llm).
```
Comment (dacorvo) [inline docs/source/guides/cache_system.mdx:25]:
```suggestion
- 1st approach: convert it to a set of [XLA](https://github.com/pytorch/xla/) subgraphs, and compile each subgraph with the neuronx compiler into a Neuron Executable File Format (NEFF) binary file (eg. training and inference of llm). 
```

File: docs/source/guides/cache_system.mdx [hunk 2]
```diff
@@ -12,24 +12,26 @@ specific language governing permissions and limitations under the License.
...
 with [`torch-neuronx`](https://github.com/aws-neuron/aws-neuron-samples/tree/master/torch-neuronx).
 
-When exporting a model, [`torch-neuronx`](https://github.com/aws-neuron/aws-neuron-samples/tree/master/torch-neuronx) will:
+When exporting a model, [`torch-neuronx`](https://github.com/aws-neuron/aws-neuron-samples/tree/master/torch-neuronx) will either:
+
+- 1st approach: convert it to a set of [XLA](https://github.com/pytorch/xla/) subgraphs, and compile each subgraph with the neuronx compiler into a Neuron Executable File Format (NEFF) binary file(eg. training and inference of llm). 
+
+Or
+
+- 2nd approach: recorde one graph, compile it with the neuronx-cc compiler to a NEFF binary file, and serialize as a TorchScript module(eg. inference of encoders, stable diffusion and seq2seq models).
```
Comment (dacorvo) [inline docs/source/guides/cache_system.mdx:29]:
```suggestion
- 2nd approach: record one graph, compile it with the neuronx-cc compiler to a NEFF binary file, and serialize as a TorchScript module (eg. inference of encoders, stable diffusion and seq2seq models).
```

File: docs/source/guides/cache_system.mdx [hunk 3]
```diff
@@ -12,24 +12,26 @@ specific language governing permissions and limitations under the License.
...
 
-- convert it to a set of [XLA](https://github.com/pytorch/xla/) subgraphs,
-- compile each subgraph with the neuronx compiler into a Neuron Executable File Format (NEFF) binary file.
+The graph recording is relatively fast, but the compilation takes a lot of time.
 
-The first step is relatively fast, but the compilation takes a lot of time.
-To avoid recompiling all NEFF files every time a model is loaded on a NeuronX host, [`torch-neuronx`](https://github.com/aws-neuron/aws-neuron-samples/tree/master/torch-neuronx)
- stores NEFF files in a local directory, usually `/var/tmp/neuron-compile-cache`.
+To avoid recompiling all NEFF files, for models adopting the first approach, every time a model is loaded on a NeuronX host, [`torch-neuronx`](https://github.com/aws-neuron/aws-neuron-samples/tree/master/torch-neuronx)
+ stores NEFF files in a local directory, usually `/var/tmp/neuron-compile-cache`. And for models adopting the second approach, the compiled Neuron models with checkpoint artiifacts (eg. config, tokenizer) will be stored by Optimum Neuron after the the first time compilation as well.
```
Comment (dacorvo) [inline docs/source/guides/cache_system.mdx:34]:
```suggestion
 stores NEFF files in a local directory, usually `/var/tmp/neuron-compile-cache`. And for models adopting the second approach, the compiled Neuron models with checkpoint artifacts (eg. config, tokenizer) will be stored by Optimum Neuron after the the first time compilation as well.
```

File: docs/source/guides/cache_system.mdx [hunk 4]
```diff
@@ -39,18 +41,18 @@ Note: we also support the creation of private, secured, remote model cache.
...
 
-If the NEFF files are cached, they will be fetched from the hub and directly loaded instead of being recompiled.
+If the NEFF files / Neuron models are cached, they will be fetched from the hub and directly loaded instead of being recompiled.
 
 ## How caching works
 
 The Optimum Neuron Cache is built on top of the [NeuronX compiler cache](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/general/arch/neuron-features/neuron-caching.html).
 
-It is important to understand that the cache operates on NEFF binaries, and not on the model itself.
+It is important to understand that the cache of training / models applying `transformers_neuronx`operates on NEFF binaries, and not on the model itself.
```
Comment (dacorvo) [inline docs/source/guides/cache_system.mdx:55]:
```suggestion
It is important to understand that the cache of training / models applying `transformers_neuronx` operates on NEFF binaries, and not on the model itself.
```

**Review by dacorvo — CHANGES_REQUESTED** [review]

File: docs/source/guides/cache_system.mdx (hunk 2 above)
Comment (michaelbenayoun) [inline docs/source/guides/cache_system.mdx:29]:
```suggestion
- 2nd approach: recorde one graph via tracing (TorchScript), compile it with the neuronx compiler to a NEFF binary file, and serialize as a TorchScript module(eg. inference of encoders, stable diffusion and seq2seq models).
```
Between the two bullet points you use "neuronx compiler" and "neuronx-cc compiler". I think the former is more accurate.

File: docs/source/guides/cache_system.mdx (hunk 3 above)
Comment (michaelbenayoun) [inline docs/source/guides/cache_system.mdx:34]:
```suggestion
 stores NEFF files in a local directory, usually `/var/tmp/neuron-compile-cache`. And for models adopting the second approach, the compiled Neuron models with checkpoint artiifacts (eg. config, tokenizer) will be stored by `optimum-neuron` after the the first time compilation as well.
```

File: docs/source/guides/cache_system.mdx (hunk 4 above)
Comment (michaelbenayoun) [inline docs/source/guides/cache_system.mdx:55]: I dont really understand this sentence. Especially the "training / models" part
```suggestion
It is important to understand that the cache of training / models applying `transformers_neuronx`operates on NEFF binaries, and not on the model itself.
```

**Comment by michaelbenayoun** [comment]
What's the status on that? @JingyaHuang do you need our reviews?

**Comment by JingyaHuang** [comment]
@michaelbenayoun I will need to update it a bit since there are some changes. Will ask for another review once done!

**Review by michaelbenayoun — APPROVED** [review]

**Pull request closed without merging**